### PR TITLE
Fix percentage going above 100%

### PR DESCRIPTION
**Pull request by fu20161234**

### CODE REVIEW

File: worker.py [hunk 1]
```diff
@@ -463,6 +463,7 @@ def run(self):
             start_worker(self.worker_no, self.points)
             return
         while self.cycle <= 3:
+            self.step = 0
```
Comment (modrzew) [inline worker.py]: This should be done in `Slave.main`, together with setting `self.seen` to 0.

**Comment by fu20161234** [comment]
Moved to where you mentioned. Should the self.cycle be incremented there too though?

**Comment by modrzew** [comment]
Thanks!

**Comment by modrzew** [comment]
Only now did I see your comment. No, I feel like cycle incrementation belongs to `Slave.run` (which runs `main` in a loop), and number of seen Pokemon is cycle-specific.

**Comment by fu20161234** [comment]
Yeah I can understand both ways which is why I didn't touch it, just wanted to let you think on it for 2.5 seconds.

Anyways, thanks for the awesome project and glad I could contribute a (tiny) bit!

**Pull request merged**